### ISSUE TITLE
Not able to run predict.py on IRMAS: a dataset for instrument recognition in musical audio signals

### ISSUE THREAD

**Issue by Sonny1314** [seth814/Audio-Classification#16]
I ran the previous version of predict.py that you are using in the video as the latest version was not showing output on the terminal. I have two problems.
1. When I run the prediction on the _IRMAS: a dataset for instrument recognition in musical audio signals_, I get an error that I am not able to understand.
![image](https://user-images.githubusercontent.com/50945527/84222593-9a42cd00-aaf5-11ea-9c5f-433191c150b4.png)
My folder structure was as follows
![image](https://user-images.githubusercontent.com/50945527/84222655-bcd4e600-aaf5-11ea-90c2-36b127171629.png)

2. and when I copied the only Acoustic guitar folder from wavfiles directory to a new directory wavfiles2 and then ran the  predict.py I got an incomplete output

![image](https://user-images.githubusercontent.com/50945527/84222773-07eef900-aaf6-11ea-949d-9a624c0fdde6.png)

**Comment by seth814** [comment]
I wonder what would happen if you...

`wav = np.asfortranarray(wav)`

between line 27 and 28 of clean.py

That second error is because classes list is probably too short and your model was trained to predict 10 classes. Assuming you changed 'wavfiles' to 'wavfiles2' in lines 20 and 22.

**Comment by seth814** [comment]
I removed the print statements for predict.py when I made the confusion matrix and ROC notebook.

`print(classes[y_pred])` after line 46 would show the predicted class for the index where probability was the highest in the output array.

**Comment by Sonny1314** [comment]
**CHANGES I MADE AFTER LEARNING FROM YOUR REPLY**
 added the line  `wav = np.asfortranarray(wav)` as you said and that took away the previous error message
### THINGS I DID AFTER MAKING THOSE CHANGES
Note: I am currently using the previous commit of predict.py. The one which you used in the video.
So, here are the things I tried 
1. I added 3 audios of the IRMAS dataset in the wavefiles/Acustic_Guitar directory. I got this error message after all the files that you have provided within the GitHub repository were done with prediction.
![image](https://user-images.githubusercontent.com/50945527/84260541-d72fb380-ab37-11ea-9f6e-f4cd85a5893f.png)

2. I also tried with the testing data of **irmas** dataset in a different directory but am getting the same error nonetheless. 

Also, the training files from the dataset are just 2 seconds long each.
 I have also used The long testing files 20 seconds in length but I am getting the same error.

**Comment by Sonny1314** [comment]
many thanks for replying me right away.

**Comment by seth814** [comment]
Yes. That looks like the dataset. That error looks like you are running the wrong tensorflow version. The tensorflow version should be 2.1.

I would say just focus on building some ML projects. Just pick something you are interested in. The new video series I'm planning will actually cover similarity search. (recommendation systems)

**Comment by Sonny1314** [comment]
Will be waiting for your future video on recommendation systems.
Also, I checked and I guess I am using correct version of TensorFlow.
![image](https://user-images.githubusercontent.com/50945527/84407639-bdba6480-ac28-11ea-9f4c-3963be6d0954.png)
![image](https://user-images.githubusercontent.com/50945527/84407737-daef3300-ac28-11ea-91b0-e7c675309d0d.png)



 So, instead of IRMAS dataset, I decided to run it with [FSDKaggle2018](https://zenodo.org/record/2552860#.XuJNvEUza01) dataset. The dataset is not divided into different folders like you have done with yours in **wavfiles** folder. Instead, it's around 9000 some audio samples in a single directory. Is there a way to separate them into their own folders so that I can train the model. Or separating them isn't required for training the model?

**Comment by Sonny1314** [comment]
Files have labels, that was mentioned on the dataset page, but how they are used is something I couldn't;t find on the internet.

I found a notebook using the dataset but but I guess his approach was a bit different than dividing audios into respective folders according to their labels, rather he used labels themselves to train.

https://www.kaggle.com/plumeria/fsdkaggle2018-challenge-and-visualization/notebook

![image](https://user-images.githubusercontent.com/50945527/84408814-3a9a0e00-ac2a-11ea-902e-b54833632db0.png)
![image](https://user-images.githubusercontent.com/50945527/84408825-3f5ec200-ac2a-11ea-9afc-28e76ed134ed.png)

**Comment by Sonny1314** [comment]
So, I checked and according to me, the TensorFlow version seems to be right. And now I am getting this error even with your dataset i.e. kaggle and the smaller one you provided in the repository when I run train.py.
I tried removing audio environment and creating it again and installing requairments.txt again with `--no-cache-dir`
but that didn't fix the issue.
I also uninstalled TensorFlow and installed it again by 
`pip install tensorflow==2.1.0`

installing TensorFlow again
![image](https://user-images.githubusercontent.com/50945527/84461585-6c42c180-ac8a-11ea-9f8c-1e777aeaf66b.png)
![image](https://user-images.githubusercontent.com/50945527/84461612-7fee2800-ac8a-11ea-9bc3-38a437e2e96c.png)

The error message I am getting
![image](https://user-images.githubusercontent.com/50945527/84461688-b0ce5d00-ac8a-11ea-8719-981a494654e6.png)

**Please help me how do I fix this. Code was running fine a couple of days ago.**

**Comment by Sonny1314** [comment]
`(audio) C:\Users\Saransh\Desktop\Audio-Classification>pip list
Package                  Version
------------------------ -------------------
absl-py                  0.9.0
astor                    0.8.1
astunparse               1.6.3
audioread                2.1.8
backcall                 0.1.0
cachetools               4.0.0
certifi                  2019.11.28
cffi                     1.14.0
chardet                  3.0.4
colorama                 0.4.3
cycler                   0.10.0
decorator                4.4.2
gast                     0.2.2
google-auth              1.11.3
google-auth-oauthlib     0.4.1
google-pasta             0.1.8
grpcio                   1.27.2
h5py                     2.10.0
idna                     2.9
ipykernel                5.2.0
ipython                  7.13.0
ipython-genutils         0.2.0
jedi                     0.16.0
joblib                   0.14.1
jupyter-client           6.1.0
jupyter-core             4.6.3
kapre                    0.1.7
Keras-Applications       1.0.8
Keras-Preprocessing      1.1.0
kiwisolver               1.1.0
librosa                  0.7.2
llvmlite                 0.31.0
Markdown                 3.2.1
matplotlib               3.1.3
numba                    0.48.0
numpy                    1.18.1
oauthlib                 3.1.0
opt-einsum               3.2.0
pandas                   1.0.1
parso                    0.6.2
pexpect                  4.8.0
pickleshare              0.7.5
Pillow                   7.0.0
pip                      20.0.2
prompt-toolkit           3.0.4
protobuf                 3.11.3
ptyprocess               0.6.0
pyasn1                   0.4.8
pyasn1-modules           0.2.8
pycparser                2.19
Pygments                 2.6.1
pyparsing                2.4.6
python-dateutil          2.8.1
python-speech-features   0.6
pytz                     2019.3
pywin32                  227
pyzmq                    19.0.0
requests                 2.23.0
requests-oauthlib        1.3.0
resampy                  0.2.2
rsa                      4.0
scikit-learn             0.22.2
scipy                    1.4.1
setuptools               47.1.1.post20200604
six                      1.14.0
SoundFile                0.10.3.post1
tensorboard              2.1.1
tensorboard-plugin-wit   1.6.0.post3
tensorflow               2.1.0
tensorflow-estimator     2.1.0
tensorflow-gpu-estimator 2.2.0
termcolor                1.1.0
tornado                  6.0.4
tqdm                     4.43.0
traitlets                4.3.3
urllib3                  1.25.8
wcwidth                  0.1.8
Werkzeug                 1.0.0
wheel                    0.34.2
wincertstore             0.2
wrapt                    1.12.0

(audio) C:\Users\Saransh\Desktop\Audio-Classification>






















`

**Comment by Sonny1314** [comment]
The error I was getting previously was due to `tensorflow-gpu-interpretor` was at version 2.2. Setting it to 2.1 did the job.
I trained the model using **FSDKaggle2018** dataset and sued just 15 different labels using both `lstm `and `conv2d` model.
![image](https://user-images.githubusercontent.com/50945527/84487314-33bbdb80-acbc-11ea-8de2-c78ff05bd262.png)

 **The accuracy was in 50's in both cases.**
![train result conv2d (2)](https://user-images.githubusercontent.com/50945527/84487521-81384880-acbc-11ea-8a6e-364ea74e7839.png)

Should I increase the epoch count to increase accuracy in the` lstm` model
or
Should I  use **_only_** verified files in the dataset to train the model?

**Comment by seth814** [comment]
I usually start by running only verified data first. Gives you a starting benchmark. You should add more convolutional layers to the model. (stack them in order) Mess around with parameter numbers. Just in general change parameters that will give better features that will be more descriptive/ useful.

**Comment by Sonny1314** [comment]
thank you.

**Comment by Sonny1314** [comment]
Thank you. I'll try with verified data and also try to add more layers and then try again.
